### DISCUSSION TITLE
midway 如何实现一个 TCP 服务？

### DISCUSSION THREAD

**Discussion by elicip** [midwayjs/midway#3996] [General / 闲聊]
目前碰到了一个和硬件设备比如二维码扫描器连接的工作，方式为扫描器作为客户端，向服务器发起TCP/IP 请求并处理服务器返回的应答。只看到midway 支持websocket，这种原始的TCP 应该如何处理？望赐教

**Comment by waitingsong** [comment]
你说的是 HTTP 请求么？

**Reply by elicip** [reply]
不是 HTTP，是原始的 TCP，直接用Node的话就是用net模块编写，但是我不知道怎么在midway中实现

**Reply by czy88840616** [reply]
这个其实跟 midway 关系不大，最简单的 require('net') 就好，你不会的应该是怎么在里面再复用 midway 的能力，这个涉及到自定义 framework

https://midwayjs.org/docs/component_development#%E8%87%AA%E5%AE%9A%E4%B9%89%E7%A4%BA%E4%BE%8B

**Reply by elicip** [reply]
@czy88840616 嗯，确实，我问的可能不是很恰当。我研究下你发的链接，谢谢
